src/Microsoft.CodeQuality.Analyzers.Exp/Core/Maintainability/ReviewSqlQueriesForSecurityVulnerabilities.cs

```diff
@@ -0,0 +1,249 @@
+                {
+                    //No warnings, as Constructor parameters in derived classes are assumed to be safe since this rule will check the constructor arguments at their call sites.
+                    return;
+                }
+
+                if (ReportDiagnosticIfNecessary(operationContext, argument.Value, invocationSyntax, constructorSymbol, containingSymbol))
+                {
+                    // Only report one warning per invocation
+                    return;
+                }
+            }
+        }
+
+        private static bool IsParameterSymbolVulnerable(IParameterSymbol parameter)
+        {
+            // Parameters might be vulnerable if "cmd" or "command" is in the name
+            return parameter != null &&
+                   (parameter.Name.IndexOf("cmd", StringComparison.OrdinalIgnoreCase) != -1 ||
+                    parameter.Name.IndexOf("command", StringComparison.OrdinalIgnoreCase) != -1);
+        }
+
+        private static bool ReportDiagnosticIfNecessary(OperationAnalysisContext operationContext,
+                                                 IOperation argumentValue,
+                                                 SyntaxNode syntax,
+                                                 ISymbol invokedSymbol,
+                                                 ISymbol containingMethod)
+        {
+            if (argumentValue.Type.SpecialType != SpecialType.System_String || !argumentValue.ConstantValue.HasValue)
+            {
+                // Review if the symbol passed to {invocation} in {method/field/constructor/etc} has user input.
+                operationContext.ReportDiagnostic(Diagnostic.Create(Rule,
+                                                                    syntax.GetLocation(),
+                                                                    invokedSymbol.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat),
+                                                                    containingMethod.Name));
+
+                return true;
+            }
+
+            return false;
+        }
+
+        private static void CheckForDbCommandAndDataAdapterImplementation(INamedTypeSymbol containingType,
+                                                                          INamedTypeSymbol iDbCommandType,
+                                                                          INamedTypeSymbol iDataAdapterType,
+                                                                          out bool implementsDbCommand,
+                                                                          out bool implementsDataCommand)
+        {
+            implementsDbCommand = false;
+            implementsDataCommand = false;
+            foreach (var @interface in containingType.AllInterfaces)
+            {
+                if (@interface.Equals(iDbCommandType))
```

Comment:
@333fred If two symbols refer to the same type/member/etc, are they guaranteed to be reference-equal? I've never researched this online before, but I always use `==` to compare symbols and it seems to work just as well as `.Equals`.

Reply:
Well, .Equals actually does reference equality on symbols. It doesn't actually matter, I'm just used to doing `Equals` on things.